cmd/discovery/cmd/send.go

```diff
@@ -32,18 +32,37 @@ This application is a tool to generate the needed files
 to quickly create a Cobra application.`,
 	Run: func(cmd *cobra.Command, args []string) {
 		fmt.Println("send called")
-		fmt.Println(*id)
 
-		discovery.SendPing(*myport, *targetid, *targetport)
+		nodeId, err := discovery.NewNodeFromString(*bootstrapNode)
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+
+		myAddr, err := net.ResolveUDPAddr("udp", *addr)
+		if err != nil {
+			fmt.Println("invalid address", err.Error())
+			return
+		}
+
+		myId, err := common.GenerateRandomAddress()
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+
+		mynode := discovery.NewNodeWithAddr(*myId, myAddr)
```

Comment:
so I suppose you try to bind the physical address myAddr with myId that is node identifier?

Reply:
yes